pkg/validate/validate.go

```diff
@@ -93,6 +94,29 @@ func validateProperties(p *api.Properties, externalOnly bool) (errs []error) {
 	}
 	errs = append(errs, validateAgentPoolProfiles(p.AgentPoolProfiles)...)
 	errs = append(errs, validateServicePrincipalProfile(p.ServicePrincipalProfile)...)
+	errs = append(errs, validateAuthProfile(p.AuthProfile)...)
+	return
+}
+
+func validateAuthProfile(ap *api.AuthProfile) (errs []error) {
+	if reflect.DeepEqual(ap, api.AuthProfile{}) {
+		errs = append(errs, fmt.Errorf("authProfile cannot be empty"))
+	}
+	if len(ap.IdentityProviders) == 0 {
```

Comment:
it should specifically be length 1 right now, right?

Reply:
Now - yes.

Reply:
is that valid for update cases if we need to inject an identity provider for SRE? (we're adding an htpasswd one right now too IIRC).